src/ipc/Kid.cc

```diff
@@ -57,15 +62,39 @@ Kid::stop(PidStatus const theExitStatus)
     assert(startTime != 0);
 
     isRunning = false;
+    stopTime = squid_curtime;
+    status = theExitStatus;
 
-    time_t stop_time;
-    time(&stop_time);
-    if ((stop_time - startTime) < fastFailureTimeLimit)
+    if ((stopTime - startTime) < fastFailureTimeLimit)
         ++badFailures;
     else
         badFailures = 0; // the failures are not "frequent" [any more]
 
-    status = theExitStatus;
+    reportStopped(); // after all state changes
+}
+
+/// describes a recently stopped kid
+void
+Kid::reportStopped() const
```

Comment:
This is _moved_ code. Any changes can be considered out of scope.

Please do not add debugs(). We do not want to duplicate the reporting code.

Please do not replace syslog() with debugs(). Many admins rely on these messages going to syslog.

This problem will disappear when we start using debugs() instead of syslog() in the daemon process _and_ copy level-0/1 messages to syslog(). That change is being discussed on squid-dev and is outside this project scope.